src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extensions.cs

```diff
@@ -0,0 +1,539 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal partial class SourceNamedTypeSymbol
+    {
+        private ExtensionInfo _lazyDeclaredExtensionInfo = ExtensionInfo.Sentinel;
+        private TypeSymbol? _lazyExtensionUnderlyingType = ErrorTypeSymbol.UnknownResultType;
+        private ImmutableArray<NamedTypeSymbol> _lazyBaseExtensions;
+
+        private class ExtensionInfo
+        {
+            public TypeSymbol? UnderlyingType;
+            public ImmutableArray<NamedTypeSymbol> BaseExtensions;
+            public bool IsExplicit;
```

Comment:
>public bool IsExplicit [](http://example.com/codeflow?start=12&length=22)

It doesn't look like this field is used in any meaningful way in ```_lazyDeclaredExtensionInfo```. For conflict analysis we can get it from syntax without storing in this structure.

Reply:
Beyond conflict analysis, we'll soon need to introduce an `IsExplicit` API on the extension type symbol (that will affect lookup rules). 
I think it'll be better to have that computed and stored for that purpose.

Reply:
> I think it'll be better to have that computed and stored for that purpose.

I think there is nothing to compute here. Whatever the first declaration says goes and that value can be stored in ```_flags``` when we create symbol. All other parts should obey.